Initial work toward new component bundle directories

lmorchard opened the pull request:
Some remaining TODOs:
- [x] Tests for new IncompatibleAddons component
- [x] Fix / move a broken ExperimentPage test based on IncompatibleAddons render
- [x] Tweak eslint rules for test modules using devDependencies
- [x] #2808 <strike>Actually migrate over some styles for the new bundles components</strike>
- [x] Clean up some commented out bits
- [x] Fix flow type validation failures
- [x] Docs based on https://gist.github.com/chuckharmston/01e5b7a16ba9b771863cbb8b2de64138

Done so far:

- Moved ExperimentRowCard and UpdateList components into subdirectories

- Extracted IncompatibleAddons as a component from ExperimentPage

- Webpack config to extract static/app/app.js.css from component imports

- Link to static/app/app.js.css styles from the static page template

- Storybook config changes to use frontend/src/app/**/{ComponentName}/stories.jsx files

- Mocha test command changes use frontend/src/app/**/{ComponentName}/tests.js

- Hacks to frontend/tasks/pages.js to ignore scss imports

- Hacks to test-setup.js to ignore scss imports

See also:
https://gist.github.com/chuckharmston/01e5b7a16ba9b771863cbb8b2de64138

Issue #2807

fzzzy commented on frontend/src/app/components/IncompatibleAddons/webpack.config.js:
```diff
@@ -0,0 +1,13 @@
+    rules: [
+      {
+        test: /\.(css|scss)$/,
+        loaders: ["style-loader", "css-loader", "sass-loader"],
```
The sass-loader docs gave something very different than this. I never would have figured this out...

lmorchard replied:
Ignore that... it didn't work and shouldn't have been there - I was editing the wrong file in the wrong dir & accidentally committed it

lmorchard replied:
FWIW, these are the changes to the main webpack config, basically from the sass-loader docs: https://github.com/mozilla/testpilot/pull/2859/files#diff-11e9f7f953edc64ba14b0cc350ae7b9d

fzzzy replied:
Yeah that changeset looks really good. I think I can actually understand how it works!

lmorchard committed:
Tweaks to support component directory bundles

- Storybook now also finds stories in frontend/src/app/**/stories.js

- `npm run test` now also finds tests in frontend/src/app/**/tests.js

- New /static/app/app.js.css stylesheet built from Sass styles imported
  by components

- Hacks to ignore .scss imports when using component modules outside
  Webpack for static page build & tests

- Update flowconfig to ignore .scss imports

- Small eslint upgrade so CLI matches gulp-eslint

Issue #2807

lmorchard committed:
Reorganize ExperimentRowCard component into bundle directory

Issue #2807

lmorchard committed:
Reorganize UpdateList component into bundle directory

Issue #2807

lmorchard committed:
Begin to extract IncompatibleAddons component from ExperimentPage container

Issue #2807

lmorchard committed:
Rename stories.jsx -> stories.js

See also:
https://github.com/airbnb/javascript/pull/985#issuecomment-239145468

fzzzy reviewed (approved):
This is going to be very nice.

lmorchard committed:
Quick & dirty revision to Storybook docs to incorporate new component bundle proposal
